scripts/run_full_pipeline.py

```diff
@@ -0,0 +1,99 @@
+import argparse
+import json
+import subprocess
+
+
+def parse_args():
+    description = """This script can be used to run the full pipeline.
+    This includes surveyor jobs queuing downloader jobs, which in turn should queue processor jobs.
+    This includes Microarray experiments, RNA-Seq experiments, and transcriptome indices.
+    This script must be run from the top level directory of the refinebio repository.!"""
+    parser = argparse.ArgumentParser(description=description)
+
+    accession_help_text = """Specify the accession_code of the experiment to survey.
+    If creating a Transcriptome Index an organism name like 'Homo
+    Sapiens' should be supplied if the organism is in the main
+    division of Ensembl. If the organism is in any other division it
+    should be specified like 'Arabidopsis thaliana, EnsemblPlants'."""
+    parser.add_argument("accession_code", help=accession_help_text)
+
+    return parser.parse_args()
+
+
+def get_job_to_run():
+    completed_command = subprocess.run(
+        ["./foreman/run_surveyor.sh", "get_job_to_be_run"],
+        stdout=subprocess.PIPE,  # capture_output=True
+    )
+
+    # The JSON output is on the last line of the output, but it has a
+    # newline as well.
+    last_line = completed_command.stdout.decode("utf-8").split("\n")[-2].strip()
+    if completed_command.returncode == 0:
+        return json.loads(last_line)
+
+
+def run_job(job):
+    job_name = job["job_name"]
+    job_id = job["job_id"]
+    print(f"Running {job_name} Job with id {job_id}!")
+
+    image_name = ""
+    if job["job_type"] == "DownloaderJob":
+        subcommand = "run_downloader_job"
+        image_name = "downloaders"
+    else:
+        subcommand = "run_processor_job"
+
+        if job_name == "AFFY_TO_PCL":
+            image_name = "affymetrix"
+        elif job_name == "SALMON":
+            image_name = "salmon"
+        elif job_name == "ILLUMINA_TO_PCL":
+            image_name == "illumina"
+        elif job_name in ["TRANSCRIPTOME_INDEX_LONG", "TRANSCRIPTOME_INDEX_SHORT"]:
+            image_name = "transcriptome"
+        elif job_name == "NO_OP":
+            image_name = "no_op"
+
+    completed_command = subprocess.check_call(
+        [
+            "./workers/run_job.sh",
+            "-i",
+            image_name,
+            subcommand,
+            f"--job-name={job_name}",
+            # job_name,
+            f"--job-id={job_id}",
+            # str(job_id),
+        ]
+    )
+
+    if completed_command != 0:
+        print(f"{job_name} Job with id {job_id} failed!")
+
+
+def survey_accession(accession_code):
+    completed_command = subprocess.check_call(
+        ["./foreman/run_surveyor.sh", "survey_all", "--accession", accession_code]
+    )
+
+    if completed_command != 0:
+        print(f"The Survey Job failed!")
+
+
+def run_full_pipeline(accession_code):
+    survey_accession(accession_code)
+
+    job_to_run = get_job_to_run()
+
+    while job_to_run:
```

Comment:
You might want to make this `while job_to_run is not None:` and then explicitly return `None` in `get_job_to_run()` if we fall through all the if statements

Reply:
There's only one if statement in `get_job_to_run`. The `get_job_to_be_run` management command has several but it has to return JSON, and `None` isn't valid JSON. Instead `get_job_to_be_run` returns a dict representing a processor job, a downloader job, or an empty dict. An empty dict `{}` is both valid JSON and falsey in python so it works here I think.

Is there any advantage to using an explicit `None` here?

Reply:
Good point, I didn't think of that. It's good as is then.